src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -106,11 +106,20 @@ public CartDTO addProductToCart(String email, String productName, int quantity)
         cart.setTotal(cart.getTotal() + productTotal);
         cart.setUserId(user.getId());
 
-        ;
+        cartRepository.save(cart);
 
-        return cartEntityToDto(cartRepository.save(cart));
+        List<String> userCartEntries = cart.getEntries().stream().map(CartEntry::getProductName).collect(Collectors.toList());
+
+        String productParam = String.join(",", userCartEntries);
```

Comment:
you could have mixed those 2 lines. instead of .collect(Collectors.toList) you could use .collect(Collectors.joining(",");

Reply:
done! good point! 👍 👍